### PR TITLE
add psnr to metrax

### PR DESCRIPTION
**Pull request by Mrigankkh**
### Summary [Issue #86 ]

- Added PSNR metric as a part of Image Metrics to metrax.
-  Reused test cases from SSIM to test PSNR
-  Added an ε-guard with a default value 0 to mirror Tensorflow behavior. TF returns inf when the images are identical (MSE==0).The ε-guard can clamp mse = max(mse,ε ) so the result is a huge but finite value, which avoids nan/inf explosions during averaging & logging. However, at the momennt, I've just initialized it to 0 to mirror TF behavior

### CODE REVIEW

File: src/metrax/image_metrics.py [hunk 1]
```diff
@@ -360,3 +360,73 @@ def from_model_output(  # type: ignore[override]
...
     return super().from_model_output(values=batch_ssim_values)
+
+@flax.struct.dataclass
+class PSNR(base.Average):
+  r"""
+  Peak Signal-to-Noise Ratio (PSNR) Metric.
+  
+    Given two images *x* and *y*, PSNR is defined as::
+
+      PSNR(x, y) = 20 * log10(max_val) - 10 * log10(MSE(x, y))
```
Comment (jshin1394) [inline src/metrax/image_metrics.py:371]: We use sphinx to generate our [readthedocs page](https://metrax.readthedocs.io/en/latest/), where docstrings are automatically rendered. Could you follow the [sphinx math syntax](https://sphinx-rtd-trial.readthedocs.io/en/latest/ext/math.html#:~:text=In%20addition%2C%20each%20single%20equation,..)? Feel free to take a peek at how other metrics define their docstrings. You can also run the following commands to render your own readthedocs page locally.

```
pip install -r ./docs/requirements.txt
sphinx-build ./docs /tmp/metrax_docs
python -m http.server --directory /tmp/metrax_docs
```

File: src/metrax/image_metrics.py [hunk 2]
```diff
@@ -360,3 +360,73 @@ def from_model_output(  # type: ignore[override]
...
+
+    where MSE is the mean‑squared error computed over all pixels and channels.
+  """ 
+  @staticmethod
```
Comment (jshin1394) [inline src/metrax/image_metrics.py:375]: nit: add extra line.

File: src/metrax/image_metrics.py [hunk 3]
```diff
@@ -360,3 +360,73 @@ def from_model_output(  # type: ignore[override]
...
+    mse = jnp.mean(jnp.square(img1 - img2), axis=(1, 2, 3))
+    mse = jnp.maximum(mse, eps)
+
+    psnr = 20.0 * jnp.log10(max_val) - 10.0 * jnp.log10(mse)
+    return psnr  
+  
+  @classmethod
+  def from_model_output(  
+      cls,
+      predictions: jnp.ndarray,   # Represents predicted images (y_pred)
```
Comment (jshin1394) [inline src/metrax/image_metrics.py:416]: nit: please remove. this is already explained in the docstring. same below.

File: src/metrax/image_metrics.py [hunk 4]
```diff
@@ -360,3 +360,73 @@ def from_model_output(  # type: ignore[override]
...
+    if img1.ndim != 4:  # (batch, H, W, C)
+          raise ValueError(
+              f"Inputs must be 4‑D (batch, height, width, channels), got {img1.ndim}‑D."
+          )
+
+      # Cast to float32 to avoid overflow/underflow issues in MSE and log.
+    img1 = img1.astype(jnp.float32)
+    img2 = img2.astype(jnp.float32)
+
+      # Mean‑squared error per image.
```
Comment (jshin1394) [inline src/metrax/image_metrics.py:406]: nit: let's make that the files are LINT checked throughout.

**Review by jshin1394 — COMMENTED** [review]
Thank you so much :)

File: src/metrax/image_metrics_test.py [hunk 5]
```diff
@@ -175,6 +175,63 @@ def test_ssim_against_tensorflow(
...
+            MAX_VAL_3,
+        ),
+        (
+            "psnr_identical_images",
+            PREDS_4_NP,
+            TARGETS_4_NP,
+            MAX_VAL_4,
+        ),
+    )
+  def test_psnr_against_tensorflow(
```
Comment (jshin1394) [inline src/metrax/image_metrics_test.py:204]: could you also add a test case for a large batch size example?

**Comment by Mrigankkh** [comment]
Apologies for the missing nnx wrapper; I'm on it. I shall go through that particular PR and fix all the docstring related issues as well. Thanks!
